### DISCUSSION TITLE
"Database error finding user" after importing users manually

### DISCUSSION THREAD

**Discussion by oskarrough** [supabase/supabase#3105] [Questions] [answered]
Hello, I'm attempting to migrate firebase users to Supabase `auth.user`. The imported data seems ok to me, but Supabase gives me `Database error finding user` when I try to, for instance, send a magic link via the admin panel.

Here's a full (redacted) auth.user after insert:

```
{
	instance_id: '00000000-0000-0000-0000-000000000000',
	id: 'xxxx-xxxx-xxxx-xxxx-....',
	aud: 'authenticated',
	role: 'authenticated'
	email: 'fake@example.com,
	encrypted_password: '.....',
	email_confirmed_at,
	invited_at,
	confirmation_token,
	confirmation_sent_at,
	recovery_token,
	recovery_sent_at,
	email_change_token_new,
	email_change,
	email_change_sent_at,
	last_sign_in_at: '2016-07-02T...',
	raw_app_meta_data,
	raw_user_meta_data,
	is_super_admin,
	created_at: '2016-07-02T...',
	updated_at: '2016-07-02T...',
	phone,
	phone_confirmed_at,
	phone_change,
	phone_change_token,
	phone_change_sent_at,
	confirmed_at,
	email_change_token_current,
	email_change_confirm_status: 0
}

```

Anything missing that would cause Supabase not find the user? I've compared with the `auth.user` schema and a user create regularly with Supabase SDK but couldn't find the mistake. Fresh eyes or knowledge appreciated!

**Comment by kachar** [comment, marked as answer]
I've stumbled upon a scenario where I had to set empty strings `''` instead of `null` values for certain fields like `*_token`. You may want to compare a manually created user vs a migrated one.

**Comment by oskarrough** [comment]
@kachar thank you! I went through the `auth.users` schema again and added every single field one by one. In the end what worked was adding the `is_super_admin: false` column. And now it works 🎉
